### PR TITLE
Improve server side assets validation display for steps [SCI-1747]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-1747

### CODE REVIEW

File: app/assets/javascripts/protocols/steps.js.erb [hunk 1]
```diff
@@ -593,21 +593,15 @@
...
-
-          $errInput = $new_form.find(".form-group.has-error")
-                               .first()
-                               .find("input");
-          renderFormError(e, $errInput);
-
-          formCallback($form);
-          applyCancelOnNew();
+        error: function(xhr) {
+          if (xhr.responseJSON['assets.file'] != null) {
```
Comment (ZmagoD) [inline app/assets/javascripts/protocols/steps.js.erb]: you don't need `!=  null` just negate the xhr.responseJSON['assets.file']
`if (! xhr.responseJSON['assets.file']) { .... }`

File: app/assets/javascripts/protocols/steps.js.erb [hunk 2]
```diff
@@ -593,21 +593,15 @@
...
+            $('#new-step-assets-group').addClass('has-error');
+            $('#new-step-assets-tab').addClass('has-error');
+            $.each(xhr.responseJSON['assets.file'], function(_, value) {
+               $('#new-step-assets-group').prepend('<span class="help-block">' + value + '</span>');
+            });
+          }
           animateSpinner(null, false);
-
-          TinyMCE.destroyAll();
```
Comment (ZmagoD) [inline app/assets/javascripts/protocols/steps.js.erb]: why you removed this?

File: app/assets/javascripts/sitewide/drag_n_drop.js.erb [hunk 3]
```diff
@@ -49,7 +49,6 @@
         var name = 'step[assets_attributes][' + index + '][file]';
         fd.append(name, droppedFiles[i]);
       }
-      droppedFiles = [];
```
Comment (ZmagoD) [inline app/assets/javascripts/sitewide/drag_n_drop.js.erb]: try to go on other task and drag in one file there

File: app/assets/javascripts/protocols/steps.js.erb (hunk 2 above)
Comment (okriuchykhin) [reply app/assets/javascripts/protocols/steps.js.erb]: Because it removes Tinymce editor after validation error

**Comment by ZmagoD** [comment]
There is still a problem when you add images and click cancel the and then open a new/edit step the previously added images remain cached :disappointed:

File: public/packs/manifest.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "application.js": "http://0.0.0.0:8080/packs/application.js",
+  "hello_react.js": "http://0.0.0.0:8080/packs/hello_react.js"
```
Comment (ZmagoD) [inline public/packs/manifest.json]: what is this?

Comment (okriuchykhin) [reply public/packs/manifest.json]: Rubbish from another branch

**Comment by ZmagoD** [comment]
Good job! :ship:

**Pull request merged**